brave/src/main/java/brave/secondary_sampling/SecondarySamplingInjector.java

```diff
@@ -53,15 +53,15 @@ static StringJoiner serializeWithSpanId(SecondarySamplingState state, boolean sa
     StringJoiner joiner = new StringJoiner(";");
     joiner.add(state.samplingKey());
 
-    String lastSpanId = state.parameter("spanId");
+    String upstreamSpanId = state.parameter("spanId");
     state.forEachParameter((key, value) -> {
       if (!"spanId".equals(key)) joiner.add(key + "=" + value);
     });
 
     if (sampled) {
       joiner.add("spanId=" + spanId);
-    } else if (lastSpanId != null) { // pass through the last span ID
-      joiner.add("spanId=" + lastSpanId);
+    } else if (spanId != null) { // pass through the upstream span ID
+      joiner.add("spanId=" + upstreamSpanId);
```

Comment:
this variable is named upstreamSpanId in the FinishedSpanHandler, so used this here. It is more important to clarify the name in code. changing the name of the propagated parameter (as ending in the "sampling" header) ended up a cure worse than disease.